src/mode.js

```diff
@@ -13,16 +13,25 @@ const chromaticNotes = ['c', 'c#', 'd', 'd#', 'e', 'f', 'f#', 'g', 'g#', 'a', 'a
  * @return {[type]}                        Returns the mode as an array, e.g. ['c3', 'd3', 'e3', 'f3', 'g3', 'a3', 'b3', 'c4']
  */
 const mode = (root, mode, octave, addRootFromNextOctave) => {
-	root = root || 'c';
-	mode = mode || 'ionian';
-	octave = octave ? Number(octave) : 3;
-	addRootFromNextOctave = addRootFromNextOctave === false ? false : true;
+	if (root.includes('|')) {
+		var args = root.split('|');
+		root = args[0];
+		mode = args[1];
+		octave = args[2];
+		addRootFromNextOctave = args[3] == 'false' ? false : true;
```

Comment:
Just make the version change in package.json on this line https://github.com/walmik/scribbletune/blob/master/package.json#L3

Once you do that and address the == comment, I ll play with this for a bit and merge it in. Soon after that, I ll create a patch version for allowing us use the scale name like this `scribble.scale('c major 4')` or `scribble.scale('c hungarian major 4')` or simply `scribble.scale('c major')`. I m aware I had suggested for the pipe separator but then realized it s not very intuitive. I m not gonna blick this PR for that change.. you ve done a lot already and it needs to be merged in! Just want you to know that I ll be updating that part of the code. I was initially wondering how could we manage with scales with 2 words like Hungarian Major or Harmonic Minor.. but now that i look at it.. the first part is the note (C, D, etc) and the last part is the octave.. basically a digit (3, 4 etc). These can be easily segregated and the remaining stuff can be assumed as the mode/scale name ;)